webhooks-with-pgstream.mdx

```diff
@@ -0,0 +1,555 @@
+      },
+      {
+        "id": "",
+        "name": "created_at",
+        "type": "timestamp without time zone",
+        "value": "2024-08-21 09:10:07.909214"
+      },
+      {
+        "id": "",
+        "name": "acked",
+        "type": "boolean",
+        "value": false
+      }
+    ],
+    "identity": null,
+    "metadata": {
+      "schema_id": null,
+      "table_pgstream_id": "",
+      "id_col_pgstream_id": null,
+      "version_col_pgstream_id": ""
+    }
+  }
+}
+```
+
+The event is more complex, but if you look under the `schema` column value, you see the newly added column:
+
+```json
+{
+  "name": "first_name",
+  "type": "text",
+  "unique": false,
+  "default": null,
+  "metadata": null,
+  "nullable": true,
+  "pgstream_id": "cr2a0enjc0j00h93rdj0-4"
+}
+```
+
+Note how you get not only the name and the type of the column, but also metadata information (is nullable, is unique, default value).
+
+Also, the `pgstream_id` value is unique and fixed for this column, regardless of the name. You can verify this by performing a rename:
+
+```sql
+ALTER TABLE users RENAME COLUMN first_name TO last_name;
+```
+
+## Conclusion
+
+Sending webhooks events from Postgres logical replication events is a non-trivial task, especially if you look at the corner cases and gotchas. However, [pgstream](https://github.com/xataio/pgstream) makes it pretty easy to get a nice, JSON-encoded, and consistent stream of events in the form of webhooks.
+
```

Comment:
I feel like writing up a more detailed **Conclusion** usually helps those who first reads the last :) How is something like this: 

```
In this guide, we demonstrated how to set up and trigger webhooks based on data changes in a PostgreSQL database using `pgstream`. We covered the initial setup, webhook configuration, and handling basic operations like inserts, updates, and deletes.

`pgstream` also offers advanced capabilities, such as sending both old and new values during updates and managing schema changes. By integrating `pgstream` into your PostgreSQL environment, you can automate workflows, trigger real-time notifications, and build more responsive applications, whether for data synchronization, auditing, or microservices communication.

```

Reply:
Nice! incorporated this as is :)